kodi_addon_checker/check_addon.py

```diff
@@ -458,18 +458,27 @@ def _check_for_existing_addon(report: Report, addon_path, all_repo_addons):
         repo_addons = all_repo_addons[branch]
 
         if addon_name in repo_addons:
-            if LooseVersion(addon_version) <= LooseVersion(repo_addons[addon_name]):
-                report.add(Record(PROBLEM, "%s addon already exist with version: %s in %s branch"
-                                  % (addon_name, repo_addons[addon_name], branch)))
-                return
-            else:
-                report.add(Record(INFORMATION, "%s addon also exists in %s branch but with version: %s"
-                                  % (addon_name, branch, repo_addons[addon_name])))
-                return
+            _check_versions(report, addon_name, branch, addon_version, repo_addons[addon_name], pr)
+            return
 
     report.add(Record(INFORMATION, "This is a new addon"))
 
 
+def _check_versions(report: Report, addon_name, branch, addon_version, repo_addons_version, pr):
+    if pr:
+        if LooseVersion(addon_version) > LooseVersion(repo_addons_version):
+            LOGGER.info("Addon version: %s while repo addon version: %s" % (addon_version, repo_addons_version))
+        else:
+            report.add(Record(PROBLEM, "Version of new addon must be higher than existing addon"))
+    else:
```

Comment:
If we're running on a whole repo without --PR it will report the dependency versions and fail the run multiple times due to problems being generated. Am I wrong?

Reply:
And repo runs most likely won't have any bumped addon versions. So it's expected for them to match the server most of the times.

Reply:
The `_check_for_existing_addon` function still needs some fixing (in a separate PR), but that's not related to this PR.
The PR does what title says and doesn't make things worse.

Reply:
Agreed